### PR TITLE
[improve][broker] Improve pulsar_topic_load_failed metric to record correct failed time

### PR DESCRIPTION
**Pull request by TakaHiR07**
### Motivation

pulsar_topic_load_failed metric may not record the correct failed time if topic load failed before execute BrokerService#createPersistentTopic

### Modifications

move to BrokerService#loadOrCreatePersistentTopic

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by nodece — APPROVED** [review]

**Review by thetumbled — APPROVED** [review]

**Comment by lhotari** [comment]
@thetumbled Before merging a PR, please take care to label the releases where the PR should also be applied. The current process works in a way where we add release labels for the next pending releases so that we remember to cherry-pick the fix to maintenance branches. We should always pick non-breaking fixes to the supported maintenance version branches. Currently it's branch-3.0 and branch-3.3. Right now this means adding labels `release/3.0.7` and `release/3.3.2` to the PR. 
During cherry-picking, these labels are used to search the PRs that need to be cherry-picked. Some of this process is explained in https://pulsar.apache.org/contribute/release-process/#cherry-picking-changes-scheduled-for-the-release .

I've been recently using these shell script functions: [https://github.com/lhotari/pulsar-contributor-toolbox/blob/33effd94d38c5d2b0ddad70[…]f034f62c425fe/functions/pulsar-contributor-toolbox-functions.sh](https://github.com/lhotari/pulsar-contributor-toolbox/blob/33effd94d38c5d2b0ddad700790f034f62c425fe/functions/pulsar-contributor-toolbox-functions.sh#L1392-L1415) and [https://github.com/lhotari/pulsar-contributor-toolbox/blob/33effd94d38c5d2b0ddad70[…]f034f62c425fe/functions/pulsar-contributor-toolbox-functions.sh](https://github.com/lhotari/pulsar-contributor-toolbox/blob/33effd94d38c5d2b0ddad700790f034f62c425fe/functions/pulsar-contributor-toolbox-functions.sh#L1443-L1454) . They help a lot in handling the cherry-picking. Slightly improved compared to what I documented in https://pulsar.apache.org/contribute/release-process/#cherry-picking-changes-scheduled-for-the-release.

**Comment by thetumbled** [comment]
> @thetumbled Before merging a PR, please take care to label the releases where the PR should also be applied. The current process works in a way where we add release labels for the next pending releases so that we remember to cherry-pick the fix to maintenance branches. We should always pick non-breaking fixes to the supported maintenance version branches. Currently it's branch-3.0 and branch-3.3. Right now this means adding labels `release/3.0.7` and `release/3.3.2` to the PR. During cherry-picking, these labels are used to search the PRs that need to be cherry-picked. Some of this process is explained in https://pulsar.apache.org/contribute/release-process/#cherry-picking-changes-scheduled-for-the-release .
> 
> I've been recently using these shell script functions: [https://github.com/lhotari/pulsar-contributor-toolbox/blob/33effd94d38c5d2b0ddad70[…]f034f62c425fe/functions/pulsar-contributor-toolbox-functions.sh](https://github.com/lhotari/pulsar-contributor-toolbox/blob/33effd94d38c5d2b0ddad700790f034f62c425fe/functions/pulsar-contributor-toolbox-functions.sh#L1392-L1415) and [https://github.com/lhotari/pulsar-contributor-toolbox/blob/33effd94d38c5d2b0ddad70[…]f034f62c425fe/functions/pulsar-contributor-toolbox-functions.sh](https://github.com/lhotari/pulsar-contributor-toolbox/blob/33effd94d38c5d2b0ddad700790f034f62c425fe/functions/pulsar-contributor-toolbox-functions.sh#L1443-L1454) . They help a lot in handling the cherry-picking. Slightly improved compared to what I documented in https://pulsar.apache.org/contribute/release-process/#cherry-picking-changes-scheduled-for-the-release.

Thanks for explaining.

**Comment by lhotari** [comment]
> Thanks for explaining.

@thetumbled The cherry-picking process isn't properly documented. There's a thread https://lists.apache.org/thread/j6qvt45rndnvjypcmqxsfmddqt41bxjv about improving and changing it.

In the current process it's sufficient to add the labels while merging. The committers actively working as release managers will handle cherry-picking and backporting and request for specific backports when a PR cannot be easily backported while resolving merge conflicts in cherry-picking. It's very common that there are merge conflicts and it's necessary to understand the tooling and process to handle the merge conflict resolution. The guide https://pulsar.apache.org/contribute/setup-git/ contains some suggestions for setting up git for maintaining Pulsar. If you are interested in the release manager role of one of the next Pulsar releases, you can suggest it on the dev mailing list or express interest on the Pulsar Slack #dev channel. I'm one of the committers who can you can also ping on Pulsar Slack #dev channel to help with various Pulsar release maintenance tasks. I'm looking for volunteers to help with these tasks.

**Pull request merged**
